tools/crm_node.c

```diff
@@ -363,26 +346,32 @@ run_controller_mainloop(uint32_t nodeid, bool list_nodes)
 }
 
 static void
-print_node_name(void)
+print_node_name(uint32_t nodeid)
 {
     int rc = pcmk_rc_ok;
+    char *node_name = NULL;
 
-    // Check environment first (i.e. when called by resource agent)
-    const char *name = getenv("OCF_RESKEY_" CRM_META "_" XML_LRM_ATTR_TARGET);
+    if (nodeid == 0) {
+        // Check environment first (i.e. when called by resource agent)
+        const char *name = getenv("OCF_RESKEY_" CRM_META "_" XML_LRM_ATTR_TARGET);
 
-    if (name != NULL) {
-        rc = out->message(out, "node-name", 0, name);
-
-    } else {
-        // Otherwise ask the controller
-        char *node_name = NULL;
+        if (name != NULL) {
+            rc = out->message(out, "node-name", 0, name);
+            exit_code = pcmk_rc2exitc(rc);
+            return;
```

Comment:
set error if not OK (also below)

Reply:
Rearranged the function a bit to only have one exit point, and that's where error gets set.

Reply:
Note that because pcmk__query_node_name calls out->error on error, setting error will result in this kind of thing in the output:
```
<pacemaker-result api-version="2.30" request="crm_node -N 3 --output-as=xml">
  <status code="68" message="Host does not exist">
    <errors>
      <error>Node is not known to cluster</error>
      <error>crm_node: Could not print node name: Node not found</error>
    </errors>
  </status>
</pacemaker-result>
```

But that's probably okay.  Two errors is better than no errors.

Reply:
Hmm, maybe we should whether error is already set then

Reply:
It looks like if there's an error in pcmk__query_node_name or pcmk__query_node_info, there's no way out of those functions without calling out->err().  So, probably what we should do is set exit_code but do not call g_set_error.  We don't use the GError object for any other purpose besides holding the error message.  If we've already got that, all we're missing is the right exit code.